components/ILIAS/LegalDocuments/classes/Table/DocumentTable.php

```diff
@@ -20,149 +20,232 @@
-            'title' => [$this->ui->txt('tbl_docs_head_title'), '', '25%'],
-            'created' => [$this->ui->txt('tbl_docs_head_created')],
-            'change' => [$this->ui->txt('tbl_docs_head_last_change')],
-            'criteria' => [$this->ui->txt('tbl_docs_head_criteria')],
-        ];
-    }
+        $uiTable = $this->ui->create()->table();
 
-    public function config(TableConfig $config): void
-    {
-        $config->setTitle($this->ui->txt('tbl_docs_title'));
-        $config->setSelectableColumns('created', 'change');
-    }
-
-    public function rows(TableSelection $select): array
-    {
-        return $this->mapSelection($this->row(...), $select);
-    }
+        if ((int) $this->user->getTimeFormat() === ilCalendarSettings::TIME_FORMAT_12) {
+            $date_format = $this->data_factory->dateFormat()->withTime12($this->user->getDateFormat());
+        } else {
+            $date_format = $this->data_factory->dateFormat()->withTime24($this->user->getDateFormat());
+        }
 
-    public function mapSelection(Closure $proc, TableSelection $select): array
-    {
-        $step = $this->step();
-        return array_map(fn($x) => $proc($x, $step()), $this->select($select));
-    }
+        $table = $uiTable->ordering(
+            $this->ui->txt('tbl_docs_title'),
+            [
+                'title' => $uiTable->column()->text($this->ui->txt('tbl_docs_head_title')),
+                'created' => $uiTable->column()->date($this->ui->txt('tbl_docs_head_created'), $date_format),
+                'change' => $uiTable->column()->date($this->ui->txt('tbl_docs_head_last_change'), $date_format),
+                'criteria' => $uiTable->column()->text($this->ui->txt('tbl_docs_head_criteria')),
+            ],
+            $this,
+            (new URI((string) $this->request->getUri()))->withParameter("cmd", "saveOrder")
+        )
+            ->withId('legalDocsTable')
+            ->withRequest($this->request);
+
+        if ($this->edit_links) {
+            $table = $table->withActions($this->buildTableActions());
+        }
 
-    public function select(TableSelection $select): array
-    {
-        return $this->repository->all();
+        return $table;
     }
 
-    public function row(Document $document, int $sorting): array
-    {
-        $render_order = $this->orderInputGui($document, $sorting);
-
-        return [
-            'order' => fn() => $render_order->render(),
-            'title' => $this->modal->create($document->content()),
-            'created' => ($this->format_date)($document->meta()->creation()->time()),
-            'change' => ($this->format_date)($document->meta()->lastModification()->time()),
-            'criteria' => $this->showCriteria($document, $this->showCriterion(...)),
-        ];
+    public function getRows(
+        OrderingRowBuilder $row_builder,
+        array $visible_column_ids
+    ): Generator {
+        foreach ($this->buildTableRows($this->records) as $row) {
```

Comment:
When you iterate over one document in `buildOrderingRows` you can drop the extra `foreach`, this will drop one indent level and one unnecessary iteration over the array.

```suggestion
        foreach ($this->records as $document) {
            yield $this->buildTableRow($document);
        }
```

In `buildTableRows` you can directly return `$row_builder->buildOrderingRow(...)`